mono/utils/mono-merp.c

```diff
@@ -206,95 +217,60 @@ parse_exception_type (const char *signal)
 	g_error ("Merp doesn't know how to handle %s\n", signal);
 }
 
-static gchar *
-mono_encode_merp_params (MERPStruct *merp)
-{
-	GString *output = g_string_new ("");
+#define FORMAT_TEXT_TO_FILE(handle, buff, sizeof_buff, ...) do { \
+	buff [0] = '\0'; \
+	g_snprintf (buff, sizeof_buff, __VA_ARGS__);	\
+	g_write (handle, buff, (guint32)strlen (buff));	\
+} while (0)
 
-	// Provided by icall
-	g_string_append_printf (output, "ApplicationBundleId: %s\n", merp->bundleIDArg);
-	g_string_append_printf (output, "ApplicationVersion: %s\n", merp->versionArg);
+static int merp_file_permissions = S_IWUSR | S_IRUSR | S_IRGRP | S_IROTH;
 
-	g_string_append_printf (output, "ApplicationBitness: %s\n", get_merp_bitness (merp->archArg));
+static gboolean
+mono_merp_write_params (MERPStruct *merp)
+{
+	int handle = g_open (merp->merpFilePath, O_TRUNC | O_WRONLY | O_CREAT, merp_file_permissions);
+	if (handle == -1) {
```

Comment:
`g_assertf (handle != -1, "Could not open MERP file at %s", merp->merpFilePath);` to trigger a double fault.

Reply:
Yeah, this is probably a critical failure. I'll add that.